docs/tutorial/resift-rentals.md

```diff
@@ -0,0 +1,2110 @@
+
+**[Section 3: Batch Loading & Pagination](#section-3-batch-loading-pagination)**</br>
+Main concepts: Fetch data in batches and merge data in the current fetch with the data from the previous fetches</br>
+Main ReSift APIs introduced: `share`, `merge`, `namespace`
+
+**[Section 4: Display Movie Info in a Movie Drawer](#section-4-display-movie-info-in-a-movie-drawer)**</br>
+Main concepts: Generate unique movie fetch instances via one fetch factory by passing in http request param</br>
+Main ReSift API introduced: `defineFetch`
+
+**[Section 5: Fetch Movie Data when Hovering over Movie Thumbnail](#section-5-fetch-movie-data-when-hovering-over-movie-thumbnail)**</br>
+Main concepts: Dispatch fetch when events fired</br>
+Main ReSift API introduced: `useDispatch`
+
+**[Section 6: Edit Movie](#section-6-edit-movie)**</br>
+Main concepts: Creating a fetch factory to update movie info and keeping that info in sync</br>
+Main ReSift API introduced: `share`
+
+**[Section 7: Create a Mock API using the ReSift HTTP Proxy](#section-7-create-a-mock-api-using-the-resift-http-proxy)**</br>
+Main concepts: Set up mock api endpoints</br>
+Main ReSift API introduced: `createHttpProxy`
+
+**If you run into any hurdle during this tutorial, please don't be hesitant to [open an issue on Github](https://github.com/justsift/resift/issues).**
+**Now let's dive in!**
+
+![dive in](https://media.giphy.com/media/1lxkgpEvs7pmlddf9D/giphy.gif)
+
+## Setup & Overview
+
+This project was bootstrapped with [create-react-app](https://create-react-app.dev/).
+
+**You can follow along by forking [the codesandbox project](https://codesandbox.io/s/resift-rentals-tutorial-section1-starter-csicp) we have set up for you.**
+
+This is what you'll see now:
+
+![](https://paper-attachments.dropbox.com/s_E201718DD88493055F02F9925295135B6353424DF01927749BEC4071F9BBE8D3_1569864556191_initial+start.png)
+
+Right now there’s nothing but a header. We’ll be writing in the `/src` folder during this tutorial. Let’s inspect our starter code in there:
+
+### Folder Structure
+
+- `src/index.js` is the entry point for our react code.
+- `src/App.js` is the base file for component imports.
+- `src/mockApi` holds our [mockApi](#mockapi) that provides data fetching endpoints.
+- `src/components` holds our [components](#components).
+
+### MockApi
+
+We have created a mock API to serve as the HTTP proxy of our app, it's at `/src/mockApi`. The data this mockApi grabs was scrapped from IMDB’s first 50 pages of movies sorted by most recent. It's not necessary to understand how to set up the mockApi to continue with the tutorial. But if you’re interested, you can head over to the [last section](#section-7-create-a-mock-api-using-the-resift-http-proxy) of this tutorial where we walked through its mechanism.
+
+To get started with the tutorial, all you need to know is that there are three endpoints with this API:
+
+1. `/genres`: returns an array of genres in this data shape:
+
+   ```js
```

Comment:
can you give me an example of data examples? do you mean listing out the actual data that gets returned, say instead of `name: string`, I would write `name: Coco`?